ui/pages/swaps/prepare-swap-page/review-quote.js

```diff
@@ -1011,18 +1012,13 @@ export default function ReviewQuote({ setReceiveToAmount }) {
             {viewQuotePriceDifferenceWarning}
             {(showInsufficientWarning || tokenBalanceUnavailable) && (
               <Box display={DISPLAY.FLEX} marginTop={2}>
-                <BannerAlert
-                  severity={SEVERITIES.INFO}
+                <NotEnoughBalance
                   title={t('notEnoughBalance')}
-                >
-                  <Text
-                    variant={TextVariant.bodyMd}
-                    as="h6"
-                    data-testid="mm-banner-alert-notification-text"
-                  >
-                    {actionableBalanceErrorMessage}
-                  </Text>
-                </BannerAlert>
+                  actionableBalanceErrorMessage={actionableBalanceErrorMessage}
+                  needsMoreGas={needsMoreGas}
+                  openBuyCryptoInPdapp={openBuyCryptoInPdapp}
+                  needsMoreGasText={t('buyMoreAsset', [nativeCurrencySymbol])}
+                />
```

Comment:
Would this work?
```
{isShowingWarning && (
          <>
            {viewQuotePriceDifferenceWarning}
            {(showInsufficientWarning || tokenBalanceUnavailable) && (
                <BannerAlert
                  title={t('notEnoughBalance')}
                  severity={Severity.Info}
                  description={actionableBalanceErrorMessage}
                  descriptionProps={{ 'data-testid': 'mm-banner-alert-notification-text' }}
                  actionButtonLabel={needsMoreGas ? t('buyMoreAsset', [nativeCurrencySymbol]) : undefined}
                  actionButtonOnClick={
                    needsMoreGas ? () => openBuyCryptoInPdapp() : undefined
                  }
                  marginTop={2}
                />
            )}
          </>
        )}
```

Reply:
My original intention with the `NotEnoughBalance` component was simply to speed up development. I pulled it out so I could make a story for it and adjust the styling more easily, rather than constantly reload the extension and retype in the parameters I needed.

Reply:
Please let me know if you want to keep the component as is, or if you want me to put it back inline with the original file.

Reply:
I understand and completetly agree that storybook speeds up development! One option is that we could create a story for `review-quote.js` and feed in the appropriate mock state using the providers to show the `BannerAlert` in a story. This prevents the need to create another component and allows for the component to be viewed in isolation. If that seems out of scope of this ticket I think the route you are taking is good. I would suggest using the `BannerAlert` props to show the action so it is consisten with other banners. I think we would need to include a period between the button and description if it were all on the same line.

Reply:
Thanks, I've updated the component as you suggested. We won't need the period if we go with your approach of using the `BannerAlert` props, since the button will appear on a separate line.